src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Unix.cs

```diff
@@ -158,39 +158,30 @@ public static byte[] ConvertAlpnProtocolListToByteArray(List<SslApplicationProto
                 {
                     context = new SafeDeleteSslContext((credential as SafeFreeSslCredentials)!, sslAuthenticationOptions);
                 }
+                else if (((SafeDeleteSslContext)context).Credentials != (credential as SafeFreeSslCredentials))
```

Comment:
I'm not sure I like the new credentials and update. This originally come from windows and OpenSSL really does not need this. I'm wondering if we can somehow make the callback logic common and still execute it here instead of going up to the loop and coming back again with new credentials?

Reply:
My main motivation was reusing the certificate selection code from `AcquireClientCredentials`, since it does some more certificate validation and processing which seemed to be missing here before. I think it should be possible to extract that code to some helper function and execute it immediately after cert selection callback.